geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommandTest.java

```diff
@@ -50,163 +50,191 @@
 import org.apache.geode.cache.query.security.UnrestrictedMethodAuthorizer;
 import org.apache.geode.distributed.ConfigurationPersistenceService;
 import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.management.internal.cli.GfshParseResult;
 import org.apache.geode.management.internal.cli.functions.AlterQueryServiceFunction;
 import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
 import org.apache.geode.test.junit.rules.GfshParserRule;
 
 public class AlterQueryServiceCommandTest {
+  private Set<DistributedMember> members;
+  private AlterQueryServiceCommand command;
+  private QueryConfigService mockQueryConfigService;
 
   @ClassRule
   public static GfshParserRule gfsh = new GfshParserRule();
 
-  private AlterQueryServiceCommand command;
-  private Set<DistributedMember> mockMemberSet;
-  private QueryConfigService mockQueryConfigService;
-
   @Before
-  @SuppressWarnings("unchecked")
   public void setup() {
+    members = new HashSet<>();
     command = spy(new AlterQueryServiceCommand());
-    mockMemberSet = mock(HashSet.class);
     mockQueryConfigService = mock(QueryConfigService.class);
+
+    doReturn(members).when(command).findMembers(null, null);
+  }
+
+  private void mockMembers(int amount, String memberNamePreffix) {
+    IntStream.range(0, amount).forEach(i -> {
+      DistributedMember mockMember = mock(DistributedMember.class);
+      when(mockMember.getName()).thenReturn(memberNamePreffix + i);
+      members.add(mockMember);
+    });
+  }
+
+  private String buildCommandString(String authorizerName, String authorizerParameters,
+      Boolean forceUpdate) {
+    StringBuilder commandBuilder = new StringBuilder(COMMAND_NAME);
+    commandBuilder.append(" --").append(AUTHORIZER_NAME).append("=").append(authorizerName);
+
+    if (authorizerParameters != null) {
+      commandBuilder.append(" --").append(AUTHORIZER_PARAMETERS).append("=")
+          .append(authorizerParameters);
+    }
+
+    if (forceUpdate != null) {
+      commandBuilder.append(" --").append(FORCE_UPDATE).append("=").append(forceUpdate);
+    }
+
+    return commandBuilder.toString();
   }
 
   @Test
-  public void commandReturnsCorrectMessageWithNoOptions() {
-    gfsh.executeAndAssertThat(command, COMMAND_NAME).statusIsError()
-        .containsOutput(NO_ARGUMENTS_MESSAGE);
+  public void commandParsesForceUpdateCorrectly() {
```

Comment:
Is this test needed here? It seems to be testing the functionality of the gfsh parser rather than the contents of the `AltherQueryServiceCommand` class.

Reply:
I wrote the test **before** actually adding the new parameter `forceUpdate` with its associated constraints (`specifiedDefaultValue = "true", unspecifiedDefaultValue = "false"`), so I'm actually testing that what I did was right.
Long story short, I used TDD for this one and I'd like to keep these tests.

Reply:
That makes sense.